prow/plank/reconciler.go

```diff
@@ -344,12 +344,23 @@ func (r *reconciler) defaultReconcile(ctx context.Context, request reconcile.Req
 		// Objects can be deleted from the API while being in our workqueue
 		return reconcile.Result{}, nil
 	}
+	originalPJ := pj.DeepCopy()
 
 	res, err := r.serializeIfNeeded(ctx, pj)
 	if IsTerminalError(err) {
 		// Unfixable cases like missing build clusters, do not return an error to prevent requeuing
-		r.log.WithError(err).WithFields(pjutil.ProwJobFields(pj)).
-			Error("Reconciliation failed with terminal error and will not be requeued")
+		log := r.log.WithError(err).WithFields(pjutil.ProwJobFields(pj))
+		log.Error("Reconciliation failed with terminal error and will not be requeued")
+		if !pj.Complete() {
+			pj.SetComplete()
+			pj.Status.State = prowv1.ErrorState
+			pj.Status.Description = fmt.Sprintf("Terminal error: %v.", err)
+			if err := r.pjClient.Patch(ctx, pj, ctrlruntimeclient.MergeFrom(originalPJ)); err != nil {
```

Comment:
I suppose a failure to Patch() here is fine, because we're going to look at it again anyway on the next reconciliation loop? Perhaps commenting as such is worth it?

Reply:
:+1: I added a comment.